lib/galaxy/datatypes/text.py

```diff
@@ -252,10 +253,26 @@ def _transform_dict_list_ids(dict_list):
                                          ('table_columns', 'columns')]:
                     try:
                         metadata_value = json_dict.get(b_name, None)
+                        if b_name == "columns" and metadata_value:
+                            mks = list(metadata_value[0]['metadata'].keys())
+                            keep_columns = {}
+                            for mk in mks:
+                                keep_columns[mk] = None
+                            for column in metadata_value:
+                                for k in column['metadata']:
+                                    if not column['metadata'][k] is None:
+                                        keep_columns[k] = 1
+                            final_list = []
+                            for k in keep_columns:
+                                if not keep_columns[k] is None:
+                                    final_list.append(k)
+                            final_list.sort()
+                            dataset.metadata.table_column_metadata_headers = final_list
                         if b_name in b_transform:
                             metadata_value = b_transform[b_name](metadata_value)
                         setattr(dataset.metadata, m_name, metadata_value)
                     except Exception:
+                        log.exception("Something in the metadata detection for biom1 went wrong: " + str(Exception))
```

Comment:
`log.exception()` already logs the latest traceback. This won't work anyway because you would need to assign the `Exception` to a variable.

```suggestion
                        log.exception("Something in the metadata detection for biom1 went wrong")
```